qiskit/optimization/algorithms/minimum_eigen_optimizer.py

```diff
@@ -11,43 +11,74 @@
 # that they have been altered from the originals.
 
 """A wrapper for minimum eigen solvers from Aqua to be used within the optimization module."""
-from typing import Optional, Any, Union, Tuple, List, cast
+from collections import namedtuple
+from typing import Optional, Any, Union, List, cast
 
 import numpy as np
+
 from qiskit.aqua.algorithms import MinimumEigensolver, MinimumEigensolverResult
 from qiskit.aqua.operators import StateFn, DictStateFn
-
 from .optimization_algorithm import (OptimizationResultStatus, OptimizationAlgorithm,
                                      OptimizationResult)
 from .. import QiskitOptimizationError
 from ..converters.quadratic_program_to_qubo import QuadraticProgramToQubo, QuadraticProgramConverter
 from ..problems.quadratic_program import QuadraticProgram, Variable
 
+RawSolutionSample = namedtuple('RawSolutionSample', 'bitstring fval probability')
+RawSolutionSample.__doc__ = """\
+Raw solution sample
+
+    bitstring: the basis sate
+    fval: the QUBO value
+    probability: the probability
+"""
+
+SolutionSample = namedtuple('SolutionSample', 'x fval probability status')
+SolutionSample.__doc__ = """\
+Solution sample interpreted for the original problem
+
+    x: the values of variables
+    fval: the objective function value of the original problem
+    probability: the probability
+    status: the status
+"""
```

Comment:
These tuples are not in the documentation, so I'm not sure if `__doc__` really works with sphinx.

Reply:
If we can use dataclass, it is easier... Anyways, I'll check if `__doc__` works with sphinx or not.

Reply:
I'm perfectly fine with data classes!

Reply:
Me too. But, Python 3.6 does not support dataclass...

Reply:
what about plain old classes then?

Reply:
I'm wondering to add https://pypi.org/project/dataclasses/ . Then we can use dataclass

Reply:
I adopt dataclasses